lib/puppet/util/posix.rb

```diff
@@ -186,7 +186,11 @@ def get_posix_value(location, id_field, field)
     end
 
     if check_value != field
-      check_value_id = get_posix_field(location, id_field, check_value)
+      begin
+        check_value_id = get_posix_field(location, id_field, check_value)
+      rescue Puppet::DevError
+        return id
+      end
```

Comment:
Could you check for the error case before calling get_posix_field so you do f have to catch deverror? Also would be good to have a test